subsys/power/power.c

```diff
@@ -116,6 +116,16 @@ enum power_states _sys_suspend(s32_t ticks)
 		 * in deep sleep mode.
 		 */
 		_sys_pm_idle_exit_notification_disable();
+#if CONFIG_DEVICE_POWER_MANAGEMENT
+	} else if (sys_pm_is_sleep_state(pm_state)) {
+		/* low power peripherals. */
+		if (sys_pm_low_power_devices()) {
+			LOG_ERR("Device enter low power state failed!");
+			sys_pm_notify_power_state_exit(pm_state);
+			pm_state = SYS_POWER_STATE_ACTIVE;
```

Comment:
This is not really an active state as it could happen after some devices were set to low power mode.
Unless we try to revert (activate) all powered down devices when error occur while traversing list in `sys_pm_low_power_devices`

Reply:
@pdunaj Thanks for the review, yes, that should be a problem. Please review the new patch. Thanks